crates/client/rpc/src/versions/admin/v0_1_0/methods/services.rs

```diff
@@ -10,61 +10,122 @@ const RESTART_INTERVAL: Duration = Duration::from_secs(5);
 #[async_trait]
 impl MadaraServicesRpcApiV0_1_0Server for Starknet {
     #[tracing::instrument(skip(self), fields(module = "Admin"))]
-    async fn service_rpc_disable(&self) -> RpcResult<bool> {
-        tracing::info!("🔌 Stopping RPC service...");
-        Ok(self.ctx.service_remove(MadaraService::Rpc))
+    async fn service_sync_disable(&self) -> RpcResult<MadaraServiceStatus> {
+        services_disable(&self.ctx, &[MadaraServiceId::L1Sync, MadaraServiceId::L2Sync], "Sync")
     }
 
     #[tracing::instrument(skip(self), fields(module = "Admin"))]
-    async fn service_rpc_enable(&self) -> RpcResult<bool> {
-        tracing::info!("🔌 Starting RPC service...");
-        Ok(self.ctx.service_add(MadaraService::Rpc))
+    async fn service_sync_enable(&self) -> RpcResult<MadaraServiceStatus> {
+        services_enable(&self.ctx, &[MadaraServiceId::L1Sync, MadaraServiceId::L2Sync], "Sync")
     }
 
     #[tracing::instrument(skip(self), fields(module = "Admin"))]
-    async fn service_rpc_restart(&self) -> RpcResult<bool> {
-        tracing::info!("🔌 Restarting RPC service...");
+    async fn service_sync_restart(&self) -> RpcResult<MadaraServiceStatus> {
+        services_restart(&self.ctx, &[MadaraServiceId::L1Sync, MadaraServiceId::L2Sync], "Sync").await
+    }
 
-        let res = self.ctx.service_remove(MadaraService::Rpc);
-        tokio::time::sleep(RESTART_INTERVAL).await;
-        self.ctx.service_add(MadaraService::Rpc);
+    #[tracing::instrument(skip(self), fields(module = "Admin"))]
+    async fn service_block_production_disable(&self) -> RpcResult<MadaraServiceStatus> {
+        services_disable(&self.ctx, &[MadaraServiceId::BlockProduction], "Block production")
+    }
 
-        tracing::info!("🔌 Restart complete (Rpc)");
+    #[tracing::instrument(skip(self), fields(module = "Admin"))]
+    async fn service_block_production_enable(&self) -> RpcResult<MadaraServiceStatus> {
+        services_enable(&self.ctx, &[MadaraServiceId::BlockProduction], "Block production")
+    }
 
-        return Ok(res);
+    #[tracing::instrument(skip(self), fields(module = "Admin"))]
+    async fn service_block_production_restart(&self) -> RpcResult<MadaraServiceStatus> {
+        services_restart(&self.ctx, &[MadaraServiceId::BlockProduction], "Block production").await
     }
 
     #[tracing::instrument(skip(self), fields(module = "Admin"))]
-    async fn service_sync_disable(&self) -> RpcResult<bool> {
-        tracing::info!("🔌 Stopping Sync service...");
+    async fn service_rpc_disable(&self) -> RpcResult<MadaraServiceStatus> {
+        services_disable(&self.ctx, &[MadaraServiceId::RpcUser], "Rpc")
```

Comment:
at this point why not just have an endpoint that takes the service to disable as argument

Reply:
Yep, planning on doing that, will be refactoring that now :+1: